pkg/runtime/framework/plugins/torch/torch.go

```diff
@@ -119,3 +154,30 @@ func (t *Torch) EnforceMLPolicy(info *runtime.Info, trainJob *trainer.TrainJob)
 
 	return nil
 }
+
+// calculateNumProcPerNode calculates the number of processes per node based on the provided resources.
+// It returns the calculated number of processes per node and a boolean indicating whether CPU resources were used.
+func calculateNumProcPerNode(
+	fallbackNumProcPerNode intstr.IntOrString, resources corev1.ResourceList, shouldUseCPU func(resources corev1.ResourceList) bool,
+) (intstr.IntOrString, bool) {
+	var defaultCPU int = 1
+	if resources != nil {
+		if shouldUseCPU(resources) {
+			cpuQ := resources[corev1.ResourceCPU]
+			if !cpuQ.IsZero() {
+				// Round up to whole cores
+				cpuQ.RoundUp(0) // Scale 0 means whole units (cores)
+
+				// Ensure at least 1 process
+				cpuValue := cpuQ.Value()
+				if cpuValue < int64(defaultCPU) {
+					cpuValue = int64(defaultCPU)
+				}
+
+				return intstr.FromInt(int(cpuValue)), true
+			}
```

Comment:
```suggestion
			return intstr.FromInt32(max(defaultCPU, cpuQ))
```

Why do we need such verification logic? Isn't this enough?
Additionally, `FromInt` is a deprecated function.